libs/telemetry/src/helpers.rs

```diff
@@ -0,0 +1,178 @@
+use super::models::TraceSpan;
+use derive_more::Display;
+use once_cell::sync::Lazy;
+use opentelemetry::propagation::Extractor;
+use opentelemetry::sdk::export::trace::SpanData;
+use opentelemetry::trace::{SpanId, TraceContextExt, TraceFlags, TraceId};
+use serde_json::{json, Value};
+use std::collections::HashMap;
+use tracing::Metadata;
+use tracing_subscriber::EnvFilter;
+
+pub static SHOW_ALL_TRACES: Lazy<bool> = Lazy::new(|| match std::env::var("PRISMA_SHOW_ALL_TRACES") {
+    Ok(enabled) => enabled.eq_ignore_ascii_case("true"),
+    Err(_) => false,
+});
+
+/// Traceparent is a remote span. It is identified by trace_id and span_id.
+///
+/// By "remote" we mean that this span was not emitted in the current process. In real life, it is
+/// either:
+///  - Emitted by the JS part of the Prisma ORM. This is true both for Accelerate (where the Rust
+///    part is deployed as a server) and for the ORM (where the Rust part is a shared library)
+///  - Never emitted at all. This happens when the `TraceParent` is created artificially from `TxId`
+///    (see `TxId::as_traceparent`). In this case, `TraceParent` is used only to correlate logs
+///    from different transaction operations - it is never used as a part of the trace
+#[derive(Display, Copy, Clone)]
+// This conforms with https://www.w3.org/TR/trace-context/#traceparent-header-field-values. Accelerate
+// relies on this behaviour.
+#[display(fmt = "00-{trace_id:032x}-{span_id:016x}-{flags:02x}")]
+pub struct TraceParent {
+    trace_id: TraceId,
+    span_id: SpanId,
+    flags: TraceFlags,
+}
+
+impl TraceParent {
+    pub fn from_remote_context(context: &opentelemetry::Context) -> Option<Self> {
+        let span = context.span();
+        let span_context = span.span_context();
+
+        if span_context.is_valid() {
+            Some(Self {
+                trace_id: span_context.trace_id(),
+                span_id: span_context.span_id(),
+                flags: span_context.trace_flags(),
+            })
+        } else {
+            None
+        }
+    }
+
+    // TODO(aqrln): remove this method once the log capturing doesn't rely on trace IDs anymore
+    #[deprecated = "this must only be used to create an artificial traceparent for log capturing when tracing is disabled on the client"]
```

Comment:
Do we have a Linear task for this?

Reply:
yeah this what I'm working on next sprint